src/Microsoft.AspNet.Mvc.Core/ModelBinding/ByteArrayModelBinder.cs

```diff
@@ -22,25 +23,26 @@ public class ByteArrayModelBinder : IModelBinder
 
             var valueProviderResult = await bindingContext.ValueProvider.GetValueAsync(bindingContext.ModelName);
 
-            // case 1: there was no <input ... /> element containing this data
+            // Check for missing data case 1: There was no <input ... /> element containing this data.
             if (valueProviderResult == null)
             {
-                return null;
+                return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
             }
 
             var value = valueProviderResult.AttemptedValue;
 
-            // case 2: there was an <input ... /> element but it was left blank
+            // Check for missing data case 2: There was an <input ... /> element but it was left blank.
             if (string.IsNullOrEmpty(value))
             {
-                return null;
+                return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
```

Comment:
This would mean that we would never be validating a controller property of type byte[] when there is no data. 
This makes it inconsistent with things like FromFileModelBinder. 
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormFileModelBinder.cs

@dougbu can we open a tracking issue, we can triage and come to a conclusion on this.